### PR TITLE
Resolve race in ShimProxyCallback::AddRef

### PR DESCRIPTION
**Pull request by benaadams**
Resolves https://github.com/dotnet/coreclr/issues/15631

### CODE REVIEW

File: src/debug/di/shimcallback.cpp [hunk 1]
```diff
@@ -31,8 +31,8 @@ ShimProxyCallback::ShimProxyCallback(ShimProcess * pShim)
 // Implement IUnknown
 ULONG ShimProxyCallback::AddRef()
 {
-    InterlockedIncrement(&m_cRef);
-    return m_cRef;
+    LONG ref = InterlockedIncrement(&m_cRef);
+    return ref;
```
Comment (mikedn) [inline src/debug/di/shimcallback.cpp]: why not `return InterlockedIncremenr(&m_cRef);`?

The race condition claim is practically bogus, if this code is changed then at least let's make it shorter/simpler.

Comment (benaadams) [reply src/debug/di/shimcallback.cpp]: Done

Comment (Dmitry-Me) [reply src/debug/di/shimcallback.cpp]: @mikedn Code you propose is shorter but it's a bit harder to see what the function returns in debugger with such code. A temporary for result makes is easier.

Comment (mikedn) [reply src/debug/di/shimcallback.cpp]: I've heard that idea many times before. The problem with is that it's impractical, it doesn't really make sense to introduce temporary variables whenever you return an expression of certain complexity. Besides, it's actually pretty easy to find out the return value in a debugger.

**Pull request merged**